Add recipe for numbex

enricoflor opened the pull request:
### Brief summary of what the package does

This package manages automatic numbering of items and references to them in a buffer. This is useful for taking notes that involve referring to items that are uniquely identified in the text by a number, as it is standard practice for linguist. With this package the numbering is automatically taken care of and so is the correct reference to each example. In other words, it automatizes a task too tedious to do manually, thereby making it easier to take notes in plain text (e.g. org mode) and export them.

### Direct link to the package repository

https://github.com/enricoflor/numbex

### Your association with the package

I am the author and maintainer.

### Relevant communications with the upstream package maintainer

None needed.

### Checklist

<!-- Please confirm with `x`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this!  Quick first pass below --

`numbex.el` with byte-compile using Emacs 27.1:
```
numbex.el:191:1:Warning: Use of deprecated ((lambda nil ...) ...) form
numbex.el:311:1:Warning: Use of deprecated ((lambda nil ...) ...) form
In end of data:
numbex.el:800:1:Warning: the function `font-lock-update' is not known to be
    defined.
```

`numbex.el` with checkdoc 0.6.2:
```
numbex.el:165: White space found at end of line
```

`numbex.el` with [melpazoid](https://github.com/riscy/melpazoid):
- numbex.el#L453: `concat` may be unneeded; `insert` can take multiple arguments
- numbex.el#L455: `concat` may be unneeded; `insert` can take multiple arguments
- numbex.el#L459: `concat` may be unneeded; `insert` can take multiple arguments
- numbex.el#L592: `concat` may be unneeded; `insert` can take multiple arguments
- numbex.el#L593: `concat` may be unneeded; `insert` can take multiple arguments
- numbex.el#L606: `concat` may be unneeded; `insert` can take multiple arguments
- numbex.el#L613: `concat` may be unneeded; `insert` can take multiple arguments

Package and license:
- Avoid specifying `:branch` except in unusual cases; MELPA checks the repo's default branch

enricoflor commented:
Thank you! I just pushed two [commits](https://github.com/enricoflor/numbex/commit/fcb45c83ae0ad053ab408551d473db4a2ae24123) that should have addressed these issues:

- Removed the unnecessary `concat`s and the white space
- Removed the deprecated use of the lambda forms
- Got rid of `font-lock-update`

Interestingly, `package-lint` on Emacs 28 wouldn't warn me about the last two.

riscy commented:
Those last two checks (`insert/concat` and `:branch`) aren't outside of package-lint at least for now -- no worries for not catching them.  And I've gone ahead and updated your recipe to remove the unnecessary `:branch` specifier as well.  I figured that was safe because there's only one branch but you can let me know here if there's something I missed.

This looks good to merge now.  And, welcome to MELPA. :)
